libs/util/src/lib/object.ts

```diff
@@ -13,17 +14,20 @@ export function crawlObject(
       )
 }
 
-//assign obj.[keys...] = value
-export function layeredAssignment<T>(
-  obj: any,
+// assign a value to a nested object, creating the path if it doesn't exist yet. obj.[keys...] = value
+export function layeredAssignment<T, Obj>(
+  obj: Obj,
   keys: readonly (number | string)[],
   value: T
-) {
+): Obj {
   keys.reduce((accu, key, i, arr) => {
-    if (i === arr.length - 1) return (accu[key] = value)
+    if (i === arr.length - 1) {
+      accu[key] = value
+      return accu
```

Comment:
This is a different return value, did you mean `accu[key]`, or has the semantic changed?

Reply:
the return value here doesn't matter, because this is the last iteration of the `reduce` function, and we are not using the returned value of the `reduce` function. returning `accu` makes sure that at all paths, the same shape is being returned back(even in a logic branch that doesn't matter"